### ISSUE TITLE
multi label support in Scala xgboost.

### ISSUE THREAD

**Issue by aniruddhghatpande** [dmlc/xgboost#10121]
I see support for multi Label objective functions for python https://xgboost.readthedocs.io/en/release_2.0.0/python/examples/custom_softmax.html.  Would there be any support for such in Scala lib or only python is currently avaiable? Exploring multi objective support for Scala package for xgboost

Also if we want to implement this in XGBoost JVM wrapper does updating the LabeledPoint Class be the only touch point in JVM wrapper or do i require to make changes on core XGBoost engine/code 

@trivialfis

**Comment by trivialfis** [comment]
cc @wbo4958

**Comment by wbo4958** [comment]
Yeah, seems its doable, but it's may be a little kind of complicated, which may touch many part of jvm packages. Anyway, please try it.

**Comment by aniruddhghatpande** [comment]
Could you confirm if this will not require any core xgboost library change ? If i add more than 1 label in jvm package would this work smoothly or requires more changes in JVM .

**Comment by wbo4958** [comment]
I think it requires more changes in JVM packages.

**Comment by aniruddhghatpande** [comment]
How could i search for python changes so i can mimic them in scala? Also how/where do i find documentation for internal implementation for multi label in xgboost? How does xgboost optimize for all labels in a target ?

**Comment by wbo4958** [comment]
At present, neither xgboost pyspark nor xgboost jvm package supports multi-labels, so there is no reference implementation for this functionality.

**Comment by trivialfis** [comment]
We have a Python native implementation that's not yet available for distributed systems. Feel free to look into the sklearn interface for multi label classifier.

**Comment by trivialfis** [comment]
Closing in favor of https://github.com/dmlc/xgboost/issues/9043 .
